File: src/main/java/com/googongill/aditory/exception/BusinessException.java
package com.googongill.aditory.exception;

import com.googongill.aditory.common.code.BusinessErrorCode;
import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public class BusinessException extends RuntimeException {

    private final BusinessErrorCode errorCode;

    public BusinessException(BusinessErrorCode errorCode) {
        super(errorCode.getMessage());
        this.errorCode = errorCode;
    }

    public BusinessException(String message, BusinessErrorCode errorCode) {
        super(message);
        this.errorCode = errorCode;
    }

    public HttpStatus getHttpStatus() {
        return errorCode.getHttpStatus();
    }
}

File: src/main/java/com/googongill/aditory/exception/BusinessException.kt
package com.googongill.aditory.exception

import com.googongill.aditory.common.code.BusinessErrorCode
import org.springframework.http.HttpStatus

open class BusinessException : RuntimeException {

    val errorCode: BusinessErrorCode

    constructor(errorCode: BusinessErrorCode) : super(errorCode.message) {
        this.errorCode = errorCode
    }

    constructor(message: String, errorCode: BusinessErrorCode) : super(message) {
        this.errorCode = errorCode
    }

    fun getHttpStatus(): HttpStatus = errorCode.httpStatus

}
